src/main/java/com/krupatek/courier/view/accountcopy/AccountCopyForm.java

```diff
@@ -363,7 +380,20 @@ public AccountCopyForm(
                             }
                         }
 
-                        destinationComboBox.setValue(null);
+                       if( resetDestination.getValue().equals("Yes"))
+                       {
+                            destinationComboBox.setValue(null);
+
+                        }
+
+                        courierSelect.addValueChangeListener(e->{
```

Comment:
Why is it added in save button action? It can be a part of normal code where we have created courierSelect ref.